keras_nlp/models/gpt2/gpt2_causal_lm.py

```diff
@@ -324,49 +340,71 @@ def next(prompt, state, index):
 
     def generate(
         self,
-        prompt,
-        max_length,
+        inputs,
+        max_length=None,
+        batch_size=None,
     ):
         """Generate text.
 
-        This method generates text based on given `prompt`. Generation will
-        continue until `max_length` is met, and all tokens generated after
-        `end_token` will be truncated. The sampling approach used can be
-        controlled via the sampler argument.
+        This method generates text based given `inputs` as a prompt.
 
         Args:
-            prompt: a string, string Tensor or string RaggedTensor. The prompt
-                text for generation.
+            inputs: a list of strings, `tf.Tensor`, or `tf.data.Dataset`.
             max_length: int. The max length of generated sequence.
             sampler: a string or `keras_nlp.samplers.Sampler` instance. The
                 sampler to be used for text generation.
         """
-        if self.preprocessor is None:
+        max_length = max_length or self.backbone.max_sequence_length
+        input_is_scalar = False
+
+        if isinstance(inputs, tf.data.Dataset) and batch_size is not None:
             raise ValueError(
-                "`self.preprocessor` is `None`, please make sure "
-                "`preprocessor` is set before calling `generate`."
+                "When `inputs` is a `tf.data.Dataset`, please do not provide "
+                "`batch_size`. Received: "
+                f"`type(batch_size)={type(batch_size)}`."
             )
 
-        # Tokenize.
-        prompt = tf.convert_to_tensor(prompt)
-        input_is_scalar = prompt.shape.rank == 0
-        prompt = prompt[tf.newaxis] if input_is_scalar else prompt
-        prompt = self.preprocessor.tokenizer(prompt)
+        if not isinstance(inputs, tf.data.Dataset):
+            if self.preprocessor is not None:
+                inputs = tf.convert_to_tensor(inputs)
+                input_is_scalar = inputs.shape.rank == 0
+                inputs = inputs[tf.newaxis] if input_is_scalar else inputs
 
-        # Pad ragged to dense tensors.
-        padded_shape = (None, max_length)
-        min_length = tf.reduce_min(prompt.row_lengths())
-        input_mask = tf.ones_like(prompt, tf.bool).to_tensor(shape=padded_shape)
-        prompt = prompt.to_tensor(shape=padded_shape)
+            if batch_size is not None:
+                inputs = tf.data.Dataset.from_tensor_slices(inputs)
+                inputs = inputs.batch(batch_size)
 
-        # Run the (possibly compiled) generate function on dense inputs.
-        generate_function = self.make_generate_function()
-        output = generate_function(prompt, input_mask, min_length)
+        def preprocess(x, y=None, sample_weight=None):
+            if self.preprocessor is not None:
+                return self.preprocessor(x, sequence_length=max_length)
```

Comment:
If we use `preprocessor` here, I would vote for setting `add_end_token=False` by default, which is only useful for chatbot model.

Reply:
Currently both `add_end_token` and `add_end_token` default to False right?